src/coreclr/jit/lclmorph.cpp

```diff
@@ -569,49 +569,58 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
     {
         GenTree* const node = *use;
 
-        if (node->OperIs(GT_IND, GT_FIELD, GT_FIELD_ADDR))
+        switch (node->gtOper)
         {
-            MorphStructField(node, user);
-        }
-        else if (node->OperIs(GT_LCL_FLD))
-        {
-            MorphLocalField(node, user);
-        }
+            case GT_IND:
+            case GT_FIELD:
+            case GT_FIELD_ADDR:
+                MorphStructField(node, user);
+                break;
+            case GT_LCL_FLD:
+                MorphLocalField(node, user);
+                assert(node->OperIsLocal());
+                __fallthrough;
```

Comment:
```suggestion
                FALLTHROUGH;
```
(We use the macro version everywhere else)

Reply:
I ended up reverting this because the case above also can potentially morph to a local, so also needs some more handling. There is a BasicBlock helper already to check for `GT_JMP` that it uses now.

Reply:
Just for my understanding: why not use `compJmpOpUsed`?

Reply:
Yes (`BasicBlock::endsWithJmpMethod` does use `compJmpOpUsed` as an early-out)